fhir-server/src/main/java/com/ibm/fhir/server/resources/FHIRResource.java

```diff
@@ -2088,22 +2092,56 @@ public Bundle doSearch(String type, String compartment, String compartmentId,
         }
     }
 
+    /**
+     * Uses the handling property from the server config and the Prefer header from the request to determine which mode the server should use.
+     * 
+     * @throws IllegalArgumentException if the mode is STRICT and the Prefer header contains an invalid handling value.
+     */
     private boolean isLenient(Map<String, String> requestProperties) {
-        boolean lenient = true;
-
-        String handlingStringValue = getHeaderValue(requestProperties, "Prefer", "handling");
-        if ("strict".equals(handlingStringValue)) {
-            lenient = false;
+        boolean isLenient = false; // Assign it to appease the compiler
+        String stringVal = FHIRConfigHelper.getStringProperty(FHIRConfiguration.PROPERTY_HANDLING, Handling.STRICT.value());
+        Handling handlingConfig = Handling.from(stringVal);
```

Comment:
up to 3 calls here... and the same 3 calls in the case statement. 

I don't think you need this if you just test handlingConfig as a string, and pass in the found Handling.LENIENT at the end.

Reply:
different calls.  this one is reading the config preference.  within the case we're reading the Prefer header handling value.

Reply:
yes and no.  you are using the value twice in a branch which might as well be one set of branches in the switch statement.

Reply:
design choice in my opinion and we can mark as resolved

Reply:
I think the only line thats repeated is `String handlingStringValue = getHeaderValue(requestProperties, HEADERNAME_PREFER, "handling");` and I didn't want to do that before the switch because we don't need it for LEIENT_ONLY or STRICT_ONLY cases.